test/unit/rules/no-heading-inside-button-test.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+const generateRuleTests = require('../../helpers/rule-test-harness');
+const ERROR_MESSAGE = require('../../../lib/rules/no-heading-inside-button').ERROR_MESSAGE;
+
+generateRuleTests({
+  name: 'no-heading-inside-button',
+
+  config: true,
+
+  good: ['<button></button>', '<button><span></span></button>', '<button><div></div></button>'],
+
+  bad: [
+    {
+      template: '<button><h1></h1></button>',
```

Comment:
can you add a case where the h1 in the button is nested in another element?

Reply:
It won't catch it.

Reply:
but I guess it probably should catch it, right?

I think we could implement this similar to what we did in https://github.com/ember-template-lint/ember-template-lint/pull/1105, and then it should catch that case too

Reply:
Confirm, we should ensure this is flagged:

```hbs
<button><div><div><h3></h3></div></div></button>
```